unix/src/machine_stats/__init__.py

```diff
@@ -283,6 +283,61 @@ def v2_playbook_on_stats(self, stats):
         self._display.display("", screen_only=True, stderr=True)
 
 
+class MeasurementsResultCallback(ResultCallback):
+    def v2_playbook_on_stats(self, stats):
+        """How to measure fields
+
+        The fields that need to be tracked can be added in the fields_to_measure list.
+        If it's a custom field, please add it to the custom_fields_to_measure list.
+        """
+
+        fields_to_measure = ["ram_used_gb"]
+        custom_fields_to_measure = ["cpu_average", "cpu_peak"]
+
+        """Process JSON payload
+
+        Go through each server in the results, and for the fields mentioned in the 
+        `fields_to_measure` or `custom_fields_to_measure`, add its measurements to the
+        transformed data.
+        """
+
+        transformed_json_payload = []
+        for server in list(self._total_results.values()):
+            for field in server:
+                # Add data (ram_used_gb) from fields_to_measure list to the transformed_json_payload list
+                if field in fields_to_measure:
+                    server_dict = {}
+                    server_dict["measurable_type"] = "server"
+                    server_dict["field_name"] = field + "_timeseries"
+                    server_dict["value"] = server[field]
+                    server_dict["measurable"] = {"host_name": server["host_name"]}
+
+                    transformed_json_payload.append(server_dict)
+
+                # Add custom fields data (cpu_average) from custom_fields_to_measure list to the transformed_json_payload list
+                elif field == "custom_fields":
+                    for custom_field in server["custom_fields"]:
+                        if custom_field in custom_fields_to_measure:
+                            server_dict = {}
+                            server_dict["measurable_type"] = "server"
+                            server_dict["field_name"] = custom_field + "_timeseries"
```

Comment:
Also, we have made adjustments to include a timestamp for when the **cpu_utilization** measurement was captured.

![image](https://user-images.githubusercontent.com/22829270/150037668-d4b02681-ca87-4634-acbf-43b24404a563.png)

We should find a way to include this timestamp in the payload.  Currently, on the API side rails create a timestamp when the record is saved. this was kind of "ok" in the past but it is definitely not an accurate representation of the time where the measurement was captured. I'm not 100% if we need to make a few adjustments on the Rails side to properly ingest the new parameter (we can figure that out soon) for now let's add it to the payload. By default, Rails will ignore any extra parameter that is not declared as permitted in the controller.

Reply:
Sure, yeah. I'll have to refresh my memory on the API side implementation too. I'll bring this up in the next standup and I'm sure @SamDesmondKing can shed more light on this one.